### PR TITLE
added missing searching algos

### LINKED ISSUE #23: Implement searching algorithms
Add these algorithms
- [x] 1. Linear Search
- [x] 2. Binary Search
- [x] 3. Jump Search
- [x] 4. Interpolation Search
- [ ] 5. Exponential Search
- [x] 6. Fibonacci Search
- [ ] 7. Recursive program to linearly search an element in a given array
- [ ] 8. Recursive function to do a substring search

Also write test cases for the same. Tests and code can be divided in two commits though.

### PR DESCRIPTION
**Pull request by ravi5175**
### Description
Algos implemented - 
1 - Exponential Search
2 - Recursive Array Search

Test Case implemented -
1 - Exponential Search Test
2 - Recursive Search Test 

Fixes #23

### Type of Change:
- Code

### How Has This Been Tested?
- All test cases (newer and previous ones) passing

### Checklist:
- [* ] I read the **CONTRIBUTING** document before making this pull request
- [ *] My code follows the style guidelines of this project
- [ *] I have performed a self-review of my own code or materials
- [ *] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ *] My change requires a change to documentation
- [ ] I have updated documentation accordingly
- [] I have updated the requirements file
- [* ] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ *] I have added tests to cover my changes
- [ *] All new and existing tests passed.

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "restructuredtext.confPath": ""
+}
```
Comment (codesankalp) [inline .vscode/settings.json]: This file is not needed.

File: dsalgo/search.py [hunk 2]
```diff
@@ -36,19 +36,25 @@ def binary_search(self):
...
+        else:
+            high = _high
+
         mid = 0
 
         while low <= high:
 
-            mid = (high + low) // 2
+            mid = int((high + low) // 2)
+            print(mid)
```
Comment (codesankalp) [inline dsalgo/search.py]: `//` will convert it into int and please remove print statement.

File: dsalgo/search.py [hunk 3]
```diff
@@ -88,7 +94,7 @@ def jump_search(self):
 
             params : array --> arr, value --> x
 
-            returns : found --> int index, not found --> -1
+            returns : found --> int index , not found --> -1
```
Comment (codesankalp) [inline dsalgo/search.py]:
```suggestion
            returns : found --> int index, not found --> -1
```

File: tests/test_search.py [hunk 4]
```diff
@@ -28,3 +28,11 @@ def test_interpolation_search(self):
...
         self.answer = sorted(self.test_array).index(self.to_search)
         self.assertEqual(self.answer, self.search.fibonacci_search())
+
+    def test_exponential_search(self):
+        self.answer = sorted(self.test_array).index(self.to_search)
+        self.assertEqual(self.answer, self.search.exponential_search())
+    
+    def test_recursive_search(self):
+        self.answer = self.test_array.index(self.to_search)
+        self.assertEqual(self.answer, self.search.recursive_search())
```
Comment (codesankalp) [inline tests/test_search.py]: add blank line.

**Review by codesankalp — CHANGES_REQUESTED** [review]

**Review by codesankalp — CHANGES_REQUESTED** [review]
Extra spaces in end of line in file. After that it will be ready to merge.

File: tests/test_search.py [hunk 5]
```diff
@@ -28,3 +28,12 @@ def test_interpolation_search(self):
...
+
```
Comment (codesankalp) [inline tests/test_search.py]:
```suggestion
        self.assertEqual(self.answer, self.search.recursive_search())

```

File: tests/test_sort.py [hunk 6]
```diff
@@ -42,3 +42,4 @@ def test_quick(self):
          self.assertEqual(self.bySort,sortedArray)
          revsortedArray=Sort(self.testArray,"merge",True)
          self.assertEqual(self.revBySort,revsortedArray)
+
```
Comment (codesankalp) [inline tests/test_sort.py]:
```suggestion
```

**Review by codesankalp — APPROVED** [review]

**Review by Aryamanz29 — APPROVED** [review]

**Pull request merged**